templates/dashboard/collection/detail.html

```diff
@@ -89,7 +89,8 @@
                   <form id="toggle-publish-form" action="{% url 'dashboard:collection-publish' pk=collection.pk %}">
                     <div class="switch" id="collection-is-published">
                       <label>
-                        <input id="toggle-publish-switch" type="checkbox" {% if collection.is_published %} checked {% endif %}>
+                        <input id="toggle-publish-switch" type="checkbox" {% if collection.is_published %} checked {% endif %}
+                        {% if site.settings.homepage_collection == collection and collection.is_published %} disabled {% endif %} >
```

Comment:
`site.settings.homepage_collection == collection and collection.is_published` - this check could be done in the view function with the result passed to the template through context. First, it is a good practice to perform database queries in the view where we can ensure that necessary prefetching is made. Second, this line is duplicated in the template.